cpp/src/arrow/flight/transport/grpc/grpc_client.cc

```diff
@@ -564,6 +568,78 @@ class GrpcResultStream : public ResultStream {
   std::unique_ptr<::grpc::ClientReader<pb::Result>> stream_;
 };
 
+// Helpers for implementing the async versions of the calls
+
+arrow::Result<FlightInfo> ToAsyncResult(const pb::FlightInfo& proto) {
+  FlightInfo::Data info_data;
+  RETURN_NOT_OK(internal::FromProto(proto, &info_data));
+  return FlightInfo(std::move(info_data));
+}
```

Comment:
Can we share this code in existing sync version of `GetFlightInfo()`?

Reply:
With a bit of juggling, yes (done)